src/sentry/services/hybrid_cloud/notifications/impl.py

```diff
@@ -142,6 +152,38 @@ def get_settings_for_user_by_projects(
             )
         ]
 
+    def get_setting_options_by_project(
+        self,
+        *,
+        user_id: int,
+        projects: Iterable[Project],
+        type: NotificationSettingEnum,
+    ) -> MutableMapping[
+        NotificationScopeEnum,
+        MutableMapping[int, MutableMapping[ExternalProviderEnum, NotificationSettingsOptionEnum]],
+    ]:
+        """
+        Returns a a mapping of project scopes settings, with projects IDs
+        mapped to a map of provider to notifications setting values.
+        """
+        user = User.objects.get(id=user_id)
+        project_to_settings: MutableMapping[
+            int, MutableMapping[ExternalProviderEnum, NotificationSettingsOptionEnum]
+        ] = defaultdict(dict)
+
+        # TODO(snigdha): This is an N+1 query. We should optimize this if there's a visible performance impact.
```

Comment:
@snigdhas can we make it take a list of projects as well?

Reply:
sure - made that change. We're getting to a point where each helper is taking the base settings and modifying the format into some type of map which is a bit clunky